huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_groups_test.go

```diff
@@ -0,0 +1,215 @@
+package apig
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccGroupsDataSource_basic(t *testing.T) {
+	var (
+		dataSourceName = "data.huaweicloud_apig_groups.filter_by_name"
+		rName          = acceptance.RandomAccResourceName()
+		dc             = acceptance.InitDataSourceCheck(dataSourceName)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGroupsDataSource_filterByName(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckOutput("is_name_filter_useful", "true"),
+					resource.TestCheckOutput("not_found_validation_pass", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccGroupsDataSource_base(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_apig_group" "test" {
+  name        = "test_%s"
```

Comment:
```suggestion
  name        = "%[2]s"
```
the `rName` already has 'test' string, the format is `tf_test_xxxxx`.

Reply:
fixed